lib/util/version.js

```diff
@@ -21,14 +22,36 @@ function resetDetectedVersion() {
   cachedDetectedReactVersion = undefined;
 }
 
+function resolveBasedir(context) {
+  let basedir;
+  if (context) {
+    const filename = context.getFilename();
+    const dirname = path.dirname(filename);
+    try {
+      if (fs.statSync(filename).isFile()) {
+        // dirname must be dir here
+        basedir = dirname;
+      }
+    } catch (err) {
+      // https://github.com/eslint/eslint/issues/11989
+      if (err.code === 'ENOTDIR') {
+        // the error code alreay indicates that dirname is a file
+        basedir = path.dirname(dirname);
+      }
+    }
+  }
+  // fallback to cwd only if basedir not resolved
+  return basedir || process.cwd();
```

Comment:
how is this any different than `let basedir = process.cwd()` at the beginning?

Reply:
`process.cwd()` will only be called when necessary.

Reply:
It costs nothing to call it, though.

Reply:
Well, less function calls seems better to me.

Reply:
then every node program would be prohibitively slow.